### DISCUSSION TITLE
Using virtualenv inside a singularity brainvisa container

### DISCUSSION THREAD

**Discussion by JoelChavas** [brainvisa/brainvisa.github.io#72] [Show and tell]
When testing a new python software using brainvisa (for example testing dependencies, etc...), it can be useful to setup a virtual environment.
To do this, on the host system (for python3, Ubuntu 18.04):
`sudo apt-get install python3-pip`
`pip3 install virtualenv`

We then enter in the singularity container:
`bv bash`

We then build the python virtual environment:
`virtualenv --python=python3 --system-site-packages venv`

To test if it works, we activate the environment and check that it can open the soma.aims library:
`. venv/bin/activate`
`python -c 'from soma import aims'`

This should give no error.

**Comment by ylep** [comment]
This should work, but I am wondering why you are installing `virtualenv` on the host? It should already be installed in the container, so you can directly enter the container with `bv bash` create the virtual environment right away.

This leads me to realize that the `python3-venv`package is missing from the images: `virtualenv` is kind of deprecated now, and we are supposed to use `python3 -m venv` instead. I will add it to the image recipes, so it should appear in the next version of the images.
